src/conversations/Conversations.ts

```diff
@@ -24,6 +24,43 @@ export default class Conversations {
    * List all conversations with the current wallet found in the network, deduped by peer address
    */
   async list(): Promise<Conversation[]> {
+    const v1Addresses = await this.getV1Addresses()
+    const topicResults = this.client.allTopics()
+    const out: Conversation[] = []
+    for (const result of topicResults) {
+      // If no context, attempt to join with existing V1 intro
+      const topics = [result.contentTopic]
+      if (!result.context) {
+        if (v1Addresses.has(result.peerAddress)) {
```

Comment:
In cases where there is both a V1 and a V2 invite for the same address, and the V2 invite has no context, merge them into a single conversation. This is the migration path for people to seamlessly be able to continue an existing conversation started from before the release of this feature.

I definitely prefer doing this merging for developers, rather than having both an old and a new conversation returned in the list and having the developer have to somehow know to merge them themselves, and also be aware that they should only write to the V2 conversation.

Reply:
I think we'll need to figure out what is the behavior of `newConversation` in this world. Are we stitching all invites with no context into one conversation?